sdk/vaa/structs.go

```diff
@@ -742,40 +755,49 @@ func (v *VAA) VerifySignatures(addresses []common.Address) bool {
 
 // VerifySignatures verifies the signature of the BatchVAA given the signer addresses.
 // Returns true if the signatures were verified successfully.
-func (v *BatchVAA) VerifySignatures(addresses []common.Address) bool {
-	return VerifySignatures(v.SigningMsg().Bytes(), v.Signatures, addresses)
+func (b *BatchVAA) VerifySignatures(addresses []common.Address) bool {
+	return VerifySignatures(b.SigningMsg().Bytes(), b.Signatures, addresses)
 }
 
 // Marshal returns the binary representation of the BatchVAA
-func (v *BatchVAA) Marshal() ([]byte, error) {
+func (b *BatchVAA) Marshal() ([]byte, error) {
 	buf := new(bytes.Buffer)
-	MustWrite(buf, binary.BigEndian, v.Version)
-	MustWrite(buf, binary.BigEndian, v.GuardianSetIndex)
+	MustWrite(buf, binary.BigEndian, b.Version)
+	MustWrite(buf, binary.BigEndian, b.GuardianSetIndex)
 
 	// Write signatures
-	MustWrite(buf, binary.BigEndian, uint8(len(v.Signatures)))
-	for _, sig := range v.Signatures {
+	MustWrite(buf, binary.BigEndian, uint8(len(b.Signatures)))
+	for _, sig := range b.Signatures {
 		MustWrite(buf, binary.BigEndian, sig.Index)
 		buf.Write(sig.Signature[:])
 	}
 
 	// Write Body
-	buf.Write(v.serializeBody())
+	buf.Write(b.serializeBody())
 
 	return buf.Bytes(), nil
 }
 
 // Serializes the body of the BatchVAA.
-func (v *BatchVAA) serializeBody() []byte {
+func (b *BatchVAA) serializeBody() []byte {
 	buf := new(bytes.Buffer)
 
-	hashes := v.ObsvHashArray()
+	hashes := b.ObsvHashArray()
 
-	MustWrite(buf, binary.BigEndian, uint8(len(hashes)))
+	numHashes := len(hashes)
+	if numHashes > MaxBatchObservations {
+		panic(fmt.Sprintf("encountered BatchVAA Hashes length outside of expected range: %d", numHashes))
```

Comment:
Is there any chance that an external source publishing too large of a batch could panic the guardian?